stm32 boards with octo or quad spi nodes rename to spi node

FRASTM opened the pull request:
Instead of adding a pre_dt_board.cmake to ignore spi node names when octospi or quaspi node also axists in the board device tree, This PR is changing the quadspi and octospi node names to **spi@...**

When building the quadspi or octospi  node  of a stm32 board with _also_ spi node, this PR will avoid the following warning :

```
CMake Warning at ./cmake/modules/dts.cmake:415 (message):
  dtc raised one or more warnings:

  
  ./build/zephyr/zephyr.dts:1272.29-1305.5:
  Warning (spi_bus_bridge): /soc/quadspi@52005000: node name for SPI buses
  should be 'spi'

  <stdout>: Warning (spi_bus_reg): Failed prerequisite 'spi_bus_bridge'
  
```

fixes [79654](https://github.com/zephyrproject-rtos/zephyr/issues/79654)

This was already done for the boards/st/b_u585i_iot02a/ and this PR extends to stm32 boards with octospi node:

- stm32h7b3i_dk
- stm32h735g_disco
- stm32l4r9i_disco
- stm32l562e_dk/

and quadspi node:

- b_l4s5i_iot01a
- stm32h745i_disco
- stm32h747i_disco
- stm32h750b_dk
- stm32l476g_disco and stm32l496g_disco

decsny reviewed (dismissed):
DT spec says to name spi bus controller nodes as `spi`, that's the reason for the warning, so that should be the fix instead of suppressing the warning

same as https://github.com/zephyrproject-rtos/zephyr/pull/78972

FRASTM committed:
dts: arm: stm32 devices with quadspi is named spi node

Fixes the CMake Warning on build/zephyr/zephyr.dts:
Warning (spi_bus_bridge): /soc/octospi@52005000: node name
for SPI buses should be 'spi'
<stdout>: Warning (spi_bus_reg): Failed prerequisite 'spi_bus_bridge'

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
dts: arm: stm32 devices with octospi is named spi node

Fixes the CMake Warning on build/zephyr/zephyr.dts:
Warning (spi_bus_bridge): /soc/octospi@52005000: node name
for SPI buses should be 'spi'
<stdout>: Warning (spi_bus_reg): Failed prerequisite 'spi_bus_bridge'

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
--> changing the quadspi or octospi node name in the stm32 soc dtsi file 
```
octospi1: spi@44021000 {
...
```
or 
```
quadspi: spi@a0001000 {
...
```

Is it suitable for you ?

erwango reviewed (changes requested):
Thanks!
Please also extend to xspi (H5).
And finally, `boards/st/b_u585i_iot02a/pre_dt_board.cmake` could also be removed.

FRASTM committed:
dts: arm: stm32 devices with xspi is named spi node

Fixes the CMake Warning on build/zephyr/zephyr.dts:
Warning (spi_bus_bridge): /soc/xspi@47001400: node name
for SPI buses should be 'spi'

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
boards: st: remove the pre_dt_board wihout with ospi node

Since the ospi@ node does not exist anymore for the stm32u5 device
there is no more need for this pre_dt_board.cmake file.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
- also for the  xspi node  of stm32H5
- removing the  boards/st/b_u585i_iot02a/pre_dt_board.cmake

FRASTM commented:
build failure with version v4.0.0-rc2 : west build -p auto -b b_u585i_iot02a/stm32u585xx/ns tests/benchmarks/sched_queues -- -DCONFIG_SCHED_DUMB=y

GeorgeCGV commented:
What stops that from being merged or simply forgotten?

GeorgeCGV commented:
@erwango ok, I thought it is "_(not related to this PR)_"
